scipy/optimize/_linprog.py

```diff
@@ -135,742 +141,1009 @@ def linprog_terse_callback(xk, **kwargs):
+                       "b) at least one decision variable is unbounded below "
+                       "and its corresponding cost is positive. ")
+        else:  # test_empty_constraint_2
+            status = 0
+            message = ("The solution was determined in presolve as there are "
+                       "no non-trivial constraints.")
+        complete = True
+        x[c < 0] = ub_mod[c < 0]
+        x[c > 0] = lb_mod[c > 0]
+        # where c is zero, set x to a finite bound or zero
+        x_zero_c = ub_mod[c == 0]
+        x_zero_c[np.isinf(x_zero_c)] = ub_mod[c == 0][np.isinf(x_zero_c)]
+        x_zero_c[np.isinf(x_zero_c)] = 0
+        x[c == 0] = x_zero_c
+        # if this is not the last step of presolve, should convert bounds back
+        # to array and return here
+
+    # *sigh* - convert bounds back to their standard form (list of tuples)
+    # again, in retrospect, numpy array would be standard form
+    lb[np.equal(lb, -np.inf)] = None
+    ub[np.equal(ub, np.inf)] = None
+    bounds = np.hstack((lb[:, np.newaxis], ub[:, np.newaxis]))
+    bounds = bounds.tolist()
+    for i, row in enumerate(bounds):
+        for j, col in enumerate(row):
+            if str(
+                    col) == "nan":  # comparing col to float("nan") and
+                                    # np.nan doesn't work. should use np.isnan
+                bounds[i][j] = None
+
+    # remove redundant (linearly dependent) rows from equality constraints
+    n_rows_A = A_eq.shape[0]
+    redundancy_warning = ("A_eq does not appear to be of full row rank. To "
+                          "improve performance, check the problem formulation "
+                          "for redundant equality constraints.")
+    if (sps.issparse(A_eq)):
+        if rr and A_eq.size > 0:  # TODO: Fast sparse rank check?
+            A_eq, b_eq, status, message = _remove_redundancy_sparse(A_eq, b_eq)
+            if A_eq.shape[0] < n_rows_A:
+                warn(redundancy_warning, OptimizeWarning)
+            if status != 0:
+                complete = True
+        return (c, c0, A_ub, b_ub, A_eq, b_eq, bounds,
+                x, undo, complete, status, message)
+
+    # This is a wild guess for which redundancy removal algorithm will be
+    # faster. More testing would be good.
+    small_nullspace = 5
+    if rr and A_eq.size > 0:
+        try:  # TODO: instead use results of first SVD in _remove_redundancy
+            rank = np.linalg.matrix_rank(A_eq)
+        except Exception:  # oh well, we'll have to go with _remove_redundancy_dense
+            rank = 0
+    if rr and A_eq.size > 0 and rank < A_eq.shape[0]:
+        warn(redundancy_warning, OptimizeWarning)
+        dim_row_nullspace = A_eq.shape[0]-rank
+        if dim_row_nullspace <= small_nullspace:
+            A_eq, b_eq, status, message = _remove_redundancy(A_eq, b_eq)
+        if dim_row_nullspace > small_nullspace or status == 4:
```

Comment:
The meaning of `status = 4`  is probably missing. So this part is quite hard to follow.